src/test_distance_functions.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include "arc_utilities/eigen_helpers.hpp"
+#include "arc_utilities/pretty_print.hpp"
+
+int main(int argc, char** argv)
+{
+    (void)argc;
+    (void)argv;
+
+
+    const Eigen::Matrix3d r1 = Eigen::Matrix3d::Identity();
+    const Eigen::Quaterniond q1(r1);
+
+    for (double theta = -M_PI; theta <= M_PI; theta += M_PI / 20)
+    {
+        Eigen::Quaterniond q2; q2 = Eigen::AngleAxisd(theta, Eigen::Vector3d::UnitX());
```

Comment:
I like the tests, but this just tests rotations about the X axis? This might miss the more interesting cases.

Reply:
We could add tests along the other axis. If we want tests for arbitrary rotations, we would need some sort of ground truth; perhaps conversion to quaternions and checking angular distance via Eigen would work as the ground truth? I think that the odds of the Eigen implementation being wrong is fairly low.

Reply:
Trusting Eigen seems fine. I support adding this test case, but I approve o the pull request without it.